File: scraper/danawa.go
// prod_tit : 상품명

// lwst_prc : 최저가 (현금도 있는데 알잘딱해야됨)

// ratingValue: 평점

// reviewCount: 평가자 수

// 	//다나와 페이지 데이터
// 	c.JSON(200, gin.H{
// 		"name":        "로지텍 MK270r (정품) (키스킨 미포함)",
// 		"price":       29700,
// 		"url":         "https://prod.danawa.com/info/?pcode=2541329",
// 		"rating":      4.6,
// 		"rater":       6683,
// 	})

package main

import (
	"fmt"
	"log"

	"github.com/antchfx/htmlquery"
	"github.com/antchfx/jsonquery"
)

// type aggregateRating struct {
// 	Rating int `json:"ratingValue"`
// 	Rater  int `json:"reviewCount"`
// }

// type offers struct {
// 	Price int `json:"lowPrice"`
// }

// type danawaWrap struct {
// 	Name            string `json:"name"`
// 	aggregateRating `json:"aggregateRating"`
// }

func main() {
	doc, err := htmlquery.LoadURL("https://prod.danawa.com/info/?pcode=2541329")
	if err != nil {
		log.Fatal(err)
	}

	rating := htmlquery.InnerText(htmlquery.FindOne(doc, "//*[@id='danawa_wrap']/script[46]"))
	fmt.Println(rating)
	doc, err := jsonquery.Parse(rating)
	if err != nil {
		panic(err)
	}
}

File: scraper/danawa.py
from xxlimited import Str
import requests
import json
from lxml import etree
from bs4 import BeautifulSoup as bs


def danawa_scraper(code):
    url = "https://prod.danawa.com/info/?pcode="

    HEADERS = ({'User-Agent':
                'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 \
                (KHTML, like Gecko) Chrome/44.0.2403.157 Safari/537.36',
                'Accept-Language': 'en-US, en;q=0.5'})

    page = requests.get(url + str(code), headers=HEADERS)

    soup = bs(page.text, "html.parser")
    dom = etree.HTML(str(soup))

    danawWrap = json.loads(
        dom.xpath("//*[@id='danawa_wrap']/script[46]")[0].text)

    delivery = soup.select_one(
        "#blog_content > div.summary_info > div.detail_summary > div.summary_left > div.lowest_area > div.lowest_list > table > tbody.card_list > tr > td.ship > span").get_text().replace(",", "").replace("원", "")

    return {
        "title": danawWrap["name"],
        "url": url + str(code),
        "imageUrl": "https://"+danawWrap["image"][0][2:],
        "price": int(danawWrap["offers"]["lowPrice"]),
        "delivery": int(delivery),
        "grade": danawWrap["aggregateRating"]["ratingValue"],
        "rater": int(danawWrap["aggregateRating"]["reviewCount"]),
    }
